executor/show.go

```diff
@@ -1852,22 +1859,79 @@ func (e *ShowExec) fetchShowTableRegions() error {
 	// Get table regions from from pd, not from regionCache, because the region cache maybe outdated.
 	var regions []regionMeta
 	if len(e.IndexName.L) != 0 {
+		// show table * index * region
 		indexInfo := tb.Meta().FindIndexByName(e.IndexName.L)
 		if indexInfo == nil {
 			return plannercore.ErrKeyDoesNotExist.GenWithStackByArgs(e.IndexName, tb.Meta().Name)
 		}
 		regions, err = getTableIndexRegions(indexInfo, physicalIDs, tikvStore, splitStore)
 	} else {
+		// show table * region
 		regions, err = getTableRegions(tb, physicalIDs, tikvStore, splitStore)
 	}
+	if err != nil {
+		return err
+	}
 
+	regionRowItem, err := e.fetchSchedulingInfo(ctx, regions, tb.Meta())
 	if err != nil {
 		return err
 	}
-	e.fillRegionsToChunk(regions)
+
+	e.fillRegionsToChunk(regionRowItem)
 	return nil
 }
 
+func (e *ShowExec) fetchSchedulingInfo(ctx context.Context, regions []regionMeta, tbInfo *model.TableInfo) ([]showTableRegionRowItem, error) {
+	scheduleState := make(map[int64]infosync.PlacementScheduleState)
+	schedulingConstraints := make(map[int64]*model.PlacementSettings)
+	regionRowItem := make([]showTableRegionRowItem, 0)
+	tblPlacement, err := e.getTablePlacement(tbInfo)
+	if err != nil {
+		return nil, err
+	}
+
+	if tbInfo.GetPartitionInfo() != nil {
+		// partitioned table
+		for _, part := range tbInfo.GetPartitionInfo().Definitions {
+			_, err = fetchScheduleState(ctx, scheduleState, part.ID)
+			if err != nil {
+				return nil, err
+			}
+			placement, err := e.getPolicyPlacement(part.PlacementPolicyRef)
+			if err != nil {
+				return nil, err
+			}
+			if placement == nil {
+				schedulingConstraints[part.ID] = tblPlacement
+			} else {
+				schedulingConstraints[part.ID] = placement
+			}
+		}
+	} else {
+		// un-partitioned table or index
+		schedulingConstraints[tbInfo.ID] = tblPlacement
+		_, err = fetchScheduleState(ctx, scheduleState, tbInfo.ID)
+		if err != nil {
+			return nil, err
+		}
+	}
+	var constraintStr string
+	for i := range regions {
+		if constraint, ok := schedulingConstraints[regions[i].physicalID]; ok && constraint != nil {
+			constraintStr = constraint.String()
+		} else {
+			constraintStr = ""
+		}
+		regionRowItem = append(regionRowItem, showTableRegionRowItem{
+			regionMeta:            regions[i],
+			schedulingConstraints: constraintStr,
+			schedulingState:       scheduleState[regions[i].physicalID].String(),
```

Comment:
What about an empty `schedulingState` if `constraintStr` is also empty?

Reply:
Nice catch! Updated!